### DISCUSSION TITLE
best practice for larger app on CF workers

### DISCUSSION THREAD

**Discussion by dcsan** [honojs/hono#2564] [General]
I have fairly large app I'm porting to CF workers and looking at hono

is it best practice to have multiple top-level entrypoint files that import only the resources you would need for a set of endpoints?

having a normal 'mono-app' index that imports and sets up all the routes would seem to be a lot of startup (and RAM) overhead for running a single API call, esp if workers are destroyed between calls.

or does cloudflare do some type of tree shaking behind the scenes for each endpoint?

when i've done this with GCP every API endpoint was effectively a separate entrypoint that handled it's own minimal imports

if I did go that way, i'm not sure how to start up the app locally in dev mode, as there would need to be many apps effectively.

https://hono.dev/guides/best-practices#building-a-larger-application

**Comment by yusukebe** [comment]
Hi @dcsan 

You can check [Service Bindings](https://developers.cloudflare.com/workers/runtime-apis/bindings/service-bindings/).

For Cloudflare Workers, as the application grows, it is a good idea to split it up by endpoint using Service Bindings.
